server/src/main/java/org/elasticsearch/common/TimeBasedUUIDGenerator.java

```diff
@@ -35,6 +35,8 @@ class TimeBasedUUIDGenerator implements UUIDGenerator {
         assert SECURE_MUNGED_ADDRESS.length == 6;
     }
 
+    private static final Base64.Encoder BASE_64_NO_PADDING = Base64.getUrlEncoder().withoutPadding();
+
     // protected for testing
     protected long currentTimeMillis() {
         return System.currentTimeMillis();
```

Comment:
Unrelated to your change: is there a reason this generator can't use the cached version of current time from ThreadPool?

Reply:
I guess it's a little bothersome code wise, because we use this thing as a static singleton and now it would have a non-static dependency on the threadpool.
Also, and  I didn't do the math here, theoretically  though probably hard in practice ... if someone sets a very long polling interval on the cache time, that would maybe introduce some chance of collision? Probably not still annoying to adjust this to depend on the pool :)